cli/command/container/run.go

```diff
@@ -295,3 +299,44 @@ func runStartContainerErr(err error) error {
 
 	return statusError
 }
+
+func parseProxyConfig(cfg *configfile.ConfigFile, host string, o opts.ListOpts) opts.ListOpts {
+	var cfgKey string
+
+	if _, ok := cfg.Proxies[host]; !ok {
+		cfgKey = "default"
+	} else {
+		cfgKey = host
+	}
+
+	config, _ := cfg.Proxies[cfgKey]
+	permitted := map[string]*string{
```

Comment:
It only stores a `string`... but as we're assigning to `*string` further down in the code, it's easier to make it `*string` here as you can't take the address of map values (e.g `&permitted["HTTP_PROXY"]`)